components/benefit_sponsors/app/controllers/benefit_sponsors/benefit_packages/benefit_packages_controller.rb

```diff
@@ -89,7 +89,9 @@ def estimated_employee_cost_details
                                                                                                                     }).value!
         respond_to do |format|
           format.html do
-            @employee_costs = Kaminari.paginate_array(@employee_costs_result[:employee_costs]).page(params[:page]).per(5)
+            reference_product_index = @employee_costs_result[:employee_costs].find_index(@employee_costs_result[:employee_costs].select{|e| e[:product_id] == @employee_costs_result[:reference_product].id}.first)
```

Comment:
can you use `detect` instead of `select.first`

Reply:
Pushed the changes @nisanthyaganti9 please review.